src/controller/ReadInteraction.h

```diff
@@ -45,11 +45,12 @@ template <typename DecodableAttributeType>
 CHIP_ERROR ReportAttribute(Messaging::ExchangeManager * exchangeMgr, EndpointId endpointId, ClusterId clusterId,
                            AttributeId attributeId, ReportAttributeParams<DecodableAttributeType> && readParams)
 {
-    app::AttributePathParams attributePath(endpointId, clusterId, attributeId);
     app::InteractionModelEngine * engine = app::InteractionModelEngine::GetInstance();
     CHIP_ERROR err                       = CHIP_NO_ERROR;
 
-    readParams.mpAttributePathParamsList    = &attributePath;
+    std::unique_ptr<app::AttributePathParams[]> readPaths(Platform::New<app::AttributePathParams>());
+    readParams.mpAttributePathParamsList    = readPaths.get();
+    readParams.mpAttributePathParamsList[0] = app::AttributePathParams(endpointId, clusterId, attributeId);
```

Comment:
This is going to mix allocators, no?  Allocating with Platform::New, but deleting with operator delete[].
How about:
```
auto readPaths = Platform::MakeUnique<app::AttributePathParams>(endpointId, clusterId, attributeId);
readParams.mpAttributePathParamsList  = readPaths.get();
```
or so.

Reply:
updated, thanks